osu.Game/Skinning/ISkinSource.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Rulesets.Objects.Types;
+using OpenTK.Graphics;
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// Provides access to skinnable elements.
+    /// </summary>
+    public interface ISkinSource
+    {
+        event Action SourceChanged;
+
+        Drawable GetDrawableComponent(string componentName);
+
+        Texture GetTexture(string componentName);
+
+        SampleChannel GetSample(string sampleName);
+
+        Color4? GetComboColour(IHasComboInformation comboObject);

```

Comment:
I don't really like this being here since not all rulesets/things that use skins are going to use `GetComboColour`. How about reversing this instead by way of:

* Add an extension method to IHasComboIndex: `Color4 GetComboColour(this IHasComboIndex combo, ISkinSource skin) => ... ?? Color4.White`
* Use `if (obj is IHasComboIndex combo) AccentColour = combo.GetComboColour(skin);`

Reply:
s/IHasComboIndex/IHasComboInformation

Reply:
would you be okay with the parameter of this method just being `int index` instead?

Reply:
sure

Reply:
Wait I think I misunderstood, of the `ISkinSource` method?

Reply:
yeah, so the interface is not passed to the method, just the index.

i figure we probably want access to these colours from other places, so would rather leave it exposed for now.

Reply:
Give an example where you'd want access to a specific index? For the editor you'll want to have access directly to the configuration, because you need to know all combo colours and not just one index.

HitObjects as discussed.

I can't think of anywhere else where this method may be used for a single index.